### PR TITLE
Fixed to support custom claims dialect

### PR DESCRIPTION
**Pull request by Kanapriya**

### CODE REVIEW

File: components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java [hunk 1]
```diff
@@ -335,7 +330,12 @@ public static TicketGrantingTicket getTicketGrantingTicket(String ticketGranting
                         requestedClaim.put(remoteClaimUri, localClaimValue);
                     }
                 }
-                return requestedClaim;
+                if(requestedClaim.isEmpty()){
```
Comment (harsha1979) [inline components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java]: What is the use of MULTI_ATTRIBUTE_SEPARATOR values here ?

File: components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/constants/CASConstants.java [hunk 2]
```diff
@@ -21,6 +21,7 @@
     public static final String CAS_CONFIG_NAME = "cas";
     public static final String CAS_FRIENDLY_NAME = "CAS Configuration";
     public static final String CAS_APP_TYPE = "hidden";
+    public static final String MULTI_ATTRIBUTE_SEPARATOR = "MultiAttributeSeparator";
```
Comment (hasinthaindrajee) [inline components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/constants/CASConstants.java]: Isn't this avaialble in somewhere else ? This constant should be available

File: components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java [hunk 3]
```diff
@@ -242,7 +242,8 @@ public static TicketGrantingTicket getTicketGrantingTicket(String ticketGranting
             throws IdentityException {
         try {
             String username = String.valueOf(result.getSubject());
-            List<String> requestedClaims = new ArrayList<String>();
+            List<String> allClaims = new ArrayList<String>();
+            Map<String, String> claimsMap = new HashMap<String, String>();
```
Comment (hasinthaindrajee) [inline components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java]: Shall we introduce a meaningful name here ? Claims map isn't very meaningful in this context. What claims do include in this map ?

File: components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/constants/CASConstants.java (hunk 2 above)
Comment (Kanapriya) [reply components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/constants/CASConstants.java]: Fixed in https://github.com/wso2-extensions/identity-inbound-auth-cas/pull/21/commits/3f5b29c4b26a91f1fdb208e6703dd96d6477e11d

File: components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java [hunk 4]
```diff
@@ -41,18 +42,14 @@
...
 import org.wso2.carbon.user.core.util.UserCoreUtil;
 import org.wso2.carbon.utils.ConfigurationContextService;
 import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
 import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (malakasilva) [inline components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java]: avoid adding * and add individual artifacts

**Review by malakasilva — APPROVED** [review]

File: components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java [hunk 5]
```diff
@@ -255,96 +250,36 @@ public static TicketGrantingTicket getTicketGrantingTicket(String ticketGranting
...
-
-            if (userStoreManager == null || !localAuthentication) {
-                Map<ClaimMapping, String> userAttributes = result.getSubject().getUserAttributes();
-                if (userAttributes != null) {
-                    for (Map.Entry<ClaimMapping, String> entry : userAttributes.entrySet()) {
-                        if (log.isDebugEnabled()) {
-                            log.debug(entry.getKey().getLocalClaim().getClaimUri() + " ==> " + entry.getValue());
-                        }
-                        claimsMap.put(entry.getKey().getLocalClaim().getClaimUri(), entry.getValue());
+            Map<ClaimMapping, String> userAttributes = result.getSubject().getUserAttributes();
```
Comment (hasinthaindrajee) [inline components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java]: Shall we add a null check for result and getSubject as well

File: components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java [hunk 6]
```diff
@@ -237,131 +239,35 @@ public static TicketGrantingTicket getTicketGrantingTicket(String ticketGranting
         }
     }
 
-    public static Map<String, String> getUserClaimValues(AuthenticationResult result, ClaimMapping[] claimMappings,
-                                                         String profile)
+    public static Map<String, String> getUserClaimValues(AuthenticationResult result, ClaimMapping[] claimMappings)
```
Comment (hasinthaindrajee) [inline components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java]: This is a public method.. We can't change apis. So don't change method api. It's fine even if it's not used from inside

File: components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java [hunk 7]
```diff
@@ -237,131 +239,35 @@ public static TicketGrantingTicket getTicketGrantingTicket(String ticketGranting
...
-                localClaimValue = localClaimValues.get(localClaimUri);
-                remoteClaimUri = claimMapping.getRemoteClaim().getClaimUri();
-
-                log.debug("getUserClaimValues: localClaimUri=" + localClaimUri + " ==> localClaimValue="
-                        + localClaimValue + " ==> remoteClaimUri=" + remoteClaimUri);
-
-                if (localClaimValue != null) {
-                    localClaimValues.remove(localClaimUri);
-                    localClaimValues.put(remoteClaimUri, localClaimValue);
+        Map<ClaimMapping, String> userAttributes = result.getSubject().getUserAttributes();
```
Comment (hasinthaindrajee) [inline components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java]: Shall we add null checks here ? Also debug logs on our checks ?

File: components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java (hunk 5 above)
Comment (Kanapriya) [reply components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/util/CASSSOUtil.java]: Fixed in https://github.com/wso2-extensions/identity-inbound-auth-cas/pull/21/commits/278dbbb011dddf4ff853c8a9a2810b26cd48c187

**Review by hasinthaindrajee — APPROVED** [review]

**Pull request merged**